bundles/org.openhab.binding.hdpowerview/README.md

```diff
@@ -1,69 +1,196 @@
+If the configuration parameters are all valid, the binding will then automatically attempt to connect to the hub.
+If the connection succeeds, the hub will indicate its status as Online, otherwise it will show an error status. 
+
+Once the hub thing has been created and successfully connected, the binding will automatically discover all shades and scenes that are in it.
+
+- For each shade discovered: the binding will create a new dedicated thing with its own channels.
+- For each scene discovered: the binding will create a new channel dynamically within the hub thing.
+
+If in the future, you add additional shades or scenes to your system, the binding will discover them too.
 
 ## Thing Configuration
 
-PowerView things should be configured via discovery - it would be difficult to configure manually as the IDs of the shades and scenes are not exposed via the dedicated app.
+### Thing Configuration for PowerView Hub
+
+| Configuration Parameter | Description   |
+|-------------------------|---------------|
+| host                    | The host name or IP address of the hub on your network. |
+| refresh                 | The number of milli-seconds between fetches of the PowerView hub's shade state (default 60'000 one minute). |
+
+### Thing Configuration for PowerView Shades
+
+PowerView shades should preferably be configured via the automatic discovery process.
+It is quite difficult to configure manually as the `id` of the shade is not exposed in the PowerView app.
 However, the configuration parameters are described below:
 
-<table>
- <tr>
-  <td><b>Thing</b></td>
-  <td><b>Configuration Parameters</b></td>
- </tr>
- <tr>
-  <td>PowerView Hub</td>
-  <td>
-   <table>
-    <tr><td><b>host</b> - the hostname or IP address of the Hub on your network.</td></tr>
-    <tr><td><b>refresh</b> - the number of milliseconds between fetches of the PowerView Hub's shade state. Defaults to 60,000 (one minute).</td></tr>
-   </table>
-  </td>
- </tr>
- <tr>
-  <td>PowerView Shade</td>
-  <td>
-   <table>
-    <tr><td><b>id</b> - the ID of the PowerView Shade on the Hub.</td></tr>
-   </table>
-  </td>
- </tr>
-</table>
+| Configuration Parameter | Description                                                   |
+|-------------------------|---------------------------------------------------------------|
+| id                      | The ID of the PowerView shade in the app. Must be an integer. |
 
 ## Channels
 
-### PowerView Shade
+### Channels for PowerView Hub
+
+The hub always has one fixed channel as below:
+
+| Channel  | Item Type | Description |
+|----------|-----------| ------------|
+| refresh  | Switch    | If you switch on this channel, it will make the hub scan all shades and update its cache state -- _see "**Refreshing the PowerView Hub Cache**" below_. Note: include `{autoupdate="false"}` in the item configuration to avoid having to reset it to off after use. |
```

Comment:
> Can this be done periodically by the binding?

Yes, I did think about doing it automatically periodically. But my thinking was as follows..

The binding polls the hub at 60 seconds interval, so one approach would be to do such a "hard refresh" on every poll. But my observations (with only three shades in my house) show that whereas a soft poll takes only milliseconds, by contrast a hard refresh takes several seconds. I am concerned that if there were 10..15 shades, then the hard refresh time would exceed the polling interval. Also a hard refresh will wake the shades out of sleep, and since many of Hunter Douglas shades are battery powered, it risks to seriously reduce battery lifetime.

For the above reasons it is clear that, if the binding were to do hard refreshes periodically, then the interval must be significantly longer than the soft polling interval. My choice therefore was to make the hard refresh interval = "infinite". i.e. to do it manually at the discretion of the user. (And the user could set up their own rule to do it periodically, if they want to).

Note that hard refreshes are only needed in the case that the user uses a wireless remote (which talks to the shades without informing the hub). If the user would only being using OpenHAB to command the shades, then the hub is always "in the loop" so hard requests are not required.

Notwithstanding the above, I am amenable to changing the approach and doing hard refreshes periodically. There would then be a Config Param for `hard refresh interval`, that could be set to 0 to disable it. => But before making such a change, I would appreciate having inputs for my "testing team" => @berland and @mstroeve => what are your thoughts?

> Beside the usability aspect, this seems more like a ThingAction rather a Channel.

I don't agree. If the hard refresh is to be done manually then it needs a UI; so it is logical to have an item linked to a channel as below..

![image](https://user-images.githubusercontent.com/893994/88576783-deeedd00-d03d-11ea-8e9d-2598e669d96c.png)

> Alternatively, the user could disable and re-enable the Thing, which would trigger a REFRESH command.

Aargh!! IMHO that would NOT be an elegant solution at all :(

Reply:
What a shame. It seems like a conceptual issue that the shades don't send their states on itself when changed. I think the periodic hard refresh option would be a good solution, if it can be disabled by the user.

> I don't agree. If the hard refresh is to be done manually then it needs a UI; so it is logical to have an item linked to a channel as below..

I don't agree, that you don't agree :) Channels are defined to have a state. Whether it should be accessible via the UI is not a pro or con for a Channel or an Action.

However, Actions/Rules can be triggered by an Item, which make them accessible via UI. Although, I agree, that the ThingActions are not really user friendly configureable currently.

Reply:
I really really really would like to keep this as a Channel. Please? Since its functionality would be identical to the Scene Channels (and the Scene Channels are not new features, but have always been implemented as Channels in prior versions of the binding). See also the comment below..

Reply:
You know what? I think I will implement this as an auto refresh function. So in that case we don't need a Channel...

Reply:
Done